DOCS-#2986: Document `modin.experimental.engines.{pandas_on_ray, pyarrow_on_ray}` (#3005)
modin-project/modin: docs/flow/modin/experimental/engines/pandas_on_ray.rst

:orphan:

Pandas-on-Ray Module Description
""""""""""""""""""""""""""""""""
High-Level Module Overview
''''''''''''''''''''''''''
This module houses experimental functionality with pandas backend and Ray
engine. This functionality is concentrated in the ``ExperimentalPandasOnRayIO`` class,
that contains methods, which extend typical pandas API to give user more flexibility
with IO operations.

Usage Guide
'''''''''''
In order to use this experimental features, just modify standard Modin import
statement as follows:

.. code-block:: python

  # import modin.pandas as pd
  import modin.experimental.pandas as pd

Implemented Operations
''''''''''''''''''''''
For now ``ExperimentalPandasOnRayIO`` implements two methods - ``read_sql`` and
``read_csv_glob``. The first one is method that allows user to use typical
``pandas.read_sql`` function extended with `Spark-like parameters
<https://spark.apache.org/docs/2.0.0/api/R/read.jdbc.html>`_ such as
``partition_column``, ``lower_bound`` and ``upper_bound``. With these parameters,

> user will be able to partition imported data in the way he need it. The second

Reviewer: ```suggestion
the user will be able to specify how to partition the imported data. The second
```

> implemented method allows to read multiple CSV files simultaneously using directory

Reviewer: ```suggestion
implemented method allows to read multiple CSV files simultaneously when a `Python Glob <https://docs.python.org/3/library/glob.html>`_ object is provided as a parameter.
```

> patterns like it is done in the `Dask API <https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.read_csv>`_.

Reviewer: ```suggestion
```

:orphan:

Pandas-on-Ray Module Description
""""""""""""""""""""""""""""""""
High-Level Module Overview
''''''''''''''''''''''''''
This module houses experimental functionality with pandas backend and Ray

> engine. This functionality is concentrated in the ``ExperimentalPandasOnRayIO`` class,

Author: Ok, links and clause with public API was added.

that contains methods, which extend typical pandas API to give user more flexibility
with IO operations.

Usage Guide
'''''''''''
In order to use this experimental features, just modify standard Modin import
statement as follows:

.. code-block:: python

  # import modin.pandas as pd
  import modin.experimental.pandas as pd

Implemented Operations
''''''''''''''''''''''

> For now ``ExperimentalPandasOnRayIO`` implements two methods - ``read_sql`` and

Author: Ok, links were added and checked.
